src/Main.elm

```diff
@@ -74,93 +74,38 @@ init _ =
 -- this is saying GotText can be a result, error, OR string?
 
 
-type AppMsg
-    = GotText (Result Http.Error String)
-
-
-type CounterMsg
+type Msg
```

Comment:
Until your app gets really nested and it's best to keep your `Msg` type as 'flat' as possible.

Reply:
Mind blown! I didn't know I could do that (putting the GotText alongside the other states for the counter).